src/midi/midi.cpp

```diff
@@ -175,6 +186,22 @@ class MidiState {
 		return note_on_tracker[NoteAddr(channel, note)];
 	}
 
+	inline void SetChannelVolume(const uint8_t channel, const uint8_t volume)
+	{
+		assert(channel <= NumMidiChannels);
+		constexpr auto max_volume = 127;
+		assert(volume <= max_volume);
```

Comment:
+1 Awesome. I like this approach to catch illegal volumes in debug-builds (because we realistically expect legal volumes are used.. and there's no need to take defensive action unless userspace is handing it over like that).